src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -490,6 +491,16 @@ def join(
             source_plan,
         )
 
+    def _table_exists(self, table_name):
```

Comment:
actually you can use https://github.com/snowflakedb/snowpark-python/blob/2fb6672b9c5851f6098a01e55e64b1d7ffc2fa84/src/snowflake/snowpark/session.py#L1565
(and please add a return type hint for it)

Reply:
Hmmm I would think the way this function works is wrong because it does not support qualified table name (e.g. `test_db.test_schema.test_table`). The database and schema need to be explicitly specified as shown in https://docs.snowflake.com/en/sql-reference/sql/show-tables.html. This might be a bug? I could change this to using `DESC` as well.

A simple repro:

```
session = Session.builder.configs(CONNECTION_PARAMETERS).create()
session.sql("create or replace table sfan_schema.test_table (id int)").collect()
print(session._table_exists('sfan_schema.test_table'))

False
```

Reply:
yea you're right. But I'm also not sure whether using desc is appropriate because we might send a failed query for the user... A hacky way to avoid this is to use self.session._conn._cursor.describe('DESC TABLE xxx'), which will not show up in user's query history.

Reply:
That is actually a good idea :)